pandas/core/indexes/datetimes.py

```diff
@@ -608,12 +610,14 @@ def _simple_new(cls, values, name=None, freq=None, tz=None,
                            dtype=dtype, **kwargs)
             values = np.array(values, copy=False)
 
-        if is_object_dtype(values):
-            return cls(values, name=name, freq=freq, tz=tz,
```

Comment:
was this just never hit?

Reply:
Yes I think it's never hit unless I'm missing something